lib/job/arFindingAidJob.class.php

```diff
@@ -125,20 +152,116 @@ public function runJob($parameters)
 
     if ($exitCode != 0)
     {
-      $this->error('Converting the EAD FO to PDF has failed.');
-
+      $this->error($this->i18n->__('Converting the EAD FO to PDF has failed.'));
       $this->logCmdOutput($output, 'ERROR(FOP)');
 
       return false;
     }
 
-    $this->info("PDF generated successfully: $pdfPath");
+    $this->info($this->i18n->__('Finding aid generated successfully: %1', array('%1' => $pdfPath)));
 
     fclose($eadFileHandle); // Will delete the tmp file
     fclose($foFileHandle);
 
-    $this->job->setStatusCompleted();
-    $this->job->save();
+    return true;
+  }
+
+  private function upload($tempPath)
+  {
+    $this->info($this->i18n->__('Uploading finding aid (%1)...', array('%1' => $this->resource->slug)));
+
+    $pdfPath = sfConfig::get('sf_web_dir') . DIRECTORY_SEPARATOR . self::getFindingAidPath($this->resource->id);
+
+    file_put_contents($pdfPath, file_get_contents($tempPath));
+
+    $this->info($this->i18n->__('Finding aid uploaded successfully: %1', array('%1' => $pdfPath)));
+
+    // Extract finding aid transcript
+    $mimeType = 'application/' . self::getFindingAidFormat();
+
+    if (!QubitDigitalObject::canExtractText($mimeType))
+    {
+      $message = $this->i18n->__('Could not obtain transcript.');
+      $this->job->addNoteText($message);
+      $this->info($message);
+
+      return true;
+    }
+
+    $this->info($this->i18n->__('Obtaining finding aid transcript...'));
+
+    $command = sprintf('pdftotext %s - 2> /dev/null', $pdfPath);
+    exec($command, $output, $status);
+
+    if ($status != 0)
+    {
+      $message = $this->i18n->__('Obtaining the transcript has failed.');
```

Comment:
Looks fine to me! Really it's "transcript" that is the unclear part (the grammar is fine), but I can't think of a better term.